modules/_labs/select/react/lib/Select.tsx

```diff
@@ -0,0 +1,342 @@
+    }
+  };
+
+  handleKeyboardShortcuts = (event: React.KeyboardEvent): void => {
+    // console.log(event.key);
+
+    const children = React.Children.toArray(this.props.children);
+    const itemCount = children.length;
+    let isShortcut = false;
+    let nextFocusedIndex = 0;
+
+    switch (event.key) {
+      case 'ArrowUp':
+      case 'ArrowDown':
+        if (!this.state.showingMenu) {
+          this.toggleMenu(true);
+        } else if (this.state.focusedItemIndex !== null) {
+          const direction = event.key === 'ArrowUp' ? -1 : 1;
+          isShortcut = true;
+          let nextIndex = this.state.focusedItemIndex + direction;
+          while (nextIndex < itemCount && nextIndex >= 0 && children[nextIndex].props.disabled) {
+            nextIndex += direction;
+          }
+          nextFocusedIndex = nextIndex < 0 ? 0 : nextIndex >= itemCount ? itemCount - 1 : nextIndex;
+          this.setState({focusedItemIndex: nextFocusedIndex});
+        }
+        break;
+
+      case 'Spacebar':
+      case ' ':
+      case 'Enter':
+        if (this.state.focusedItemIndex !== null) {
+          const child = children[this.state.focusedItemIndex] as React.ReactElement<
+            SelectOptionProps
+          >;
+          this.handleOptionClick(child.props);
+          isShortcut = true;
+        }
+        break;
+
+      default:
+    }
+
+    if (isShortcut) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  };
+
+  renderChildren = (
+    child: React.ReactElement<SelectOptionProps>,
+    index: number
+  ): React.ReactNode => {
+    return React.cloneElement(child, {
+      focused: this.state.focusedItemIndex === index,
+      justSelected: this.state.justSelectedItemIndex === index,
+      onMouseDown: (event: React.MouseEvent) => {
+        event.preventDefault();
+        this.handleOptionClick(child.props);
+      },
+      selected: this.state.selectedItemIndex === index,
+    });
+  };
+
+  public render() {
+    // TODO: Standardize on prop spread location (see #150)
+    // IMPORTANT: Destructure value here so it doesn't override
+    // value below (which is set to label) with the ...elemProps
+    // spread
+    const {error, disabled, grow, children, onChange, value, ...elemProps} = this.props;
+
+    const {label} = this.state;
+    // console.log('in Select render with label', label);
+
+    return (
+      <SelectWrapper grow={grow} disabled={disabled}>
+        <SelectInput
+          disabled={disabled}
+          grow={grow}
+          error={error}
+          value={label}
+          onBlur={this.handleSelectBlur}
+          onChange={onChange}
+          onClick={this.handleSelectClick}
+          onKeyDown={this.handleKeyboardShortcuts}
+          readOnly
+          ref={this.inputRef}
+          type="text"
+          {...elemProps}
+        />
+        {this.state.showingMenu && (
+          <SelectMenu>{React.Children.map(children, this.renderChildren)}</SelectMenu>
+        )}
```

Comment:
There are several instances where existential UI like this should always be classified as a `Popup` and be portalled to the bottom of the `body` tag. We can provide a prop to opt-out of this feature.

The common problem is popups like this are constrained by container elements. If the select menu shows below the Select input and it is he bottom of a container, it will increase the size of the container but not scroll, causing the menu to be invisible to the user. If the `Select` is inside a Modal with an overflow container, the select menu will not be able to show outside the borders of the Modal